--- upstream/library_browser.js
+++ owned/library_browser.js
@@ -110,7 +110,7 @@
         assert(typeof url == 'string', 'createObjectURL must return a url as a string');
 #endif
         var audio = new Audio();
-        audio.addEventListener('canplaythrough', () => finish(audio), false); // use addEventListener due to chromium bug 124926
+        Module['classicListen'](audio, 'canplaythrough', () => finish(audio), false); // use addEventListener due to chromium bug 124926
         audio.onerror = function audio_onerror(event) {
           if (done) return;
           err(`warning: browser could not fully decode audio ${name}, trying slower base64 approach`);
@@ -175,13 +175,13 @@
                                  (() => {}); // no-op if function does not exist
         canvas.exitPointerLock = canvas.exitPointerLock.bind(document);
 
-        document.addEventListener('pointerlockchange', pointerLockChange, false);
-        document.addEventListener('mozpointerlockchange', pointerLockChange, false);
-        document.addEventListener('webkitpointerlockchange', pointerLockChange, false);
-        document.addEventListener('mspointerlockchange', pointerLockChange, false);
+        Module['classicListen'](document, 'pointerlockchange', pointerLockChange, false);
+        Module['classicListen'](document, 'mozpointerlockchange', pointerLockChange, false);
+        Module['classicListen'](document, 'webkitpointerlockchange', pointerLockChange, false);
+        Module['classicListen'](document, 'mspointerlockchange', pointerLockChange, false);
 
         if (Module['elementPointerLock']) {
-          canvas.addEventListener("click", (ev) => {
+          Module['classicListen'](canvas, "click", (ev) => {
             if (!Browser.pointerLock && Module['canvas'].requestPointerLock) {
               Module['canvas'].requestPointerLock();
               ev.preventDefault();
@@ -285,10 +285,10 @@
 
       if (!Browser.fullscreenHandlersInstalled) {
         Browser.fullscreenHandlersInstalled = true;
-        document.addEventListener('fullscreenchange', fullscreenChange, false);
-        document.addEventListener('mozfullscreenchange', fullscreenChange, false);
-        document.addEventListener('webkitfullscreenchange', fullscreenChange, false);
-        document.addEventListener('MSFullscreenChange', fullscreenChange, false);
+        Module['classicListen'](document, 'fullscreenchange', fullscreenChange, false);
+        Module['classicListen'](document, 'mozfullscreenchange', fullscreenChange, false);
+        Module['classicListen'](document, 'webkitfullscreenchange', fullscreenChange, false);
+        Module['classicListen'](document, 'MSFullscreenChange', fullscreenChange, false);
       }
 
       // create a new parent to ensure the canvas has no siblings. this allows browsers to optimize full screen performance when its parent is the full screen root
